src/graph/impl/KokkosGraph_Distance2MIS_impl.hpp

```diff
@@ -396,7 +396,7 @@ struct D2_MIS_RandomPriority
     Kokkos::parallel_for(range_pol(0, numVerts), InitWorklistFunctor(colWorklist));
     worklist_t thirdWorklist = Kokkos::subview(allWorklists, Kokkos::ALL(), 2);
     auto execSpaceEnum = KokkosKernels::Impl::kk_get_exec_space_type<exec_space>();
-    bool useTeams = execSpaceEnum == KokkosKernels::Impl::Exec_CUDA;
+    bool useTeams = (execSpaceEnum == KokkosKernels::Impl::Exec_CUDA) && (entries.extent(0) / numVerts >= 16);
```

Comment:
Average degree >= 16 ? Is this what you want ? It will not use the team policy for several use cases such as problems with 7 nnz per row.

Reply:
@srajama1 I believe that is what we want, at least based on the graphs in the table. bodyy, StocF and McAlister Momentum were the graphs in the table with < 16 nnz/row. Using teams was 2% faster, 16% slower, and 15% slower respectively. So overall using RangePolicy for those is faster.

Reply:
Alright. Thanks for checking !